source/NVDAObjects/IAccessible/ia2Web.py

```diff
@@ -48,6 +51,32 @@ class Editor(Ia2Web):
 class EditorChunk(Ia2Web):
 	beTransparentToMouse = True
 
+class Math(Ia2Web):
+
+	def _get_mathMl(self):
+		from comtypes.gen.ISimpleDOM import ISimpleDOMNode
+		try:
+			node = self.IAccessibleObject.QueryInterface(ISimpleDOMNode)
+			# Try the data-mathml attribute.
+			attr = node.attributesForNames(1, (BSTR * 1)("data-mathml"), (c_short * 1)(0,))

```

Comment:
It's partly knowing the ISimpleDOMNode interface (for which there is an idl) and partly knowing ctypes/comtypes. attributesForNames can retrieve multiple attributes. It takes three arguments: an attribute count, an array of BSTRs specifying the attribute names and an array of shorts specifying the namespace id for each attribute (which for our purposes is 0). Because comtypes doesn't quite understand that we're dealing with arrays here, we need to explicitly create the arrays. `BSTR * 1` creates a type for an array of BSTR with length 1 (akin to writing `BSTR foo[1]` in C++). Similar situation for `c_short * 1`; it's akin to `short foo[1]` in C++.

I guess I could simplify this by defining these as variables first, but it seemed like a pointless waste of variables at the time. The question is whether this is just something that requires knowledge of ctypes/comtypes. Does the following actually make it understandable to you? If so, I'm happy to change it.

```
attrNames = (BSTR * 1)("data-mathml")
namespaceIds = (c_short * 1)(0)
attrVal = node.attributesForNames(1, attrNames, namespaceIds)
```